pythonforandroid/build.py

```diff
@@ -737,110 +741,118 @@ def run_pymodules_install(ctx, modules, project_dir=None,
-        info('No Python modules and no setup.py to process, skipping')
-        return
 
-    # Output messages about what we're going to do:
-    if modules:
-        info('The requirements ({}) don\'t have recipes, attempting to '
-             'install them with pip'.format(', '.join(modules)))
-        info('If this fails, it may mean that the module has compiled '
-             'components and needs a recipe.')
-    if project_dir is not None and \
-            project_has_setup_py(project_dir) and not ignore_setup_py:
-        info('Will process project install, if it fails then the '
-             'project may not be compatible for Android install.')
-
-    # Use our hostpython to create the virtualenv
-    host_python = sh.Command(ctx.hostpython)
-    with current_directory(join(ctx.build_dir)):
-        shprint(host_python, '-m', 'venv', 'venv')
-
-        # Prepare base environment and upgrade pip:
-        base_env = dict(copy.copy(os.environ))
-        base_env["PYTHONPATH"] = ctx.get_site_packages_dir()
-        info('Upgrade pip to latest version')
-        shprint(sh.bash, '-c', (
-            "source venv/bin/activate && pip install -U pip"
-        ), _env=copy.copy(base_env))
-
-        # Install Cython in case modules need it to build:
-        info('Install Cython in case one of the modules needs it to build')
-        shprint(sh.bash, '-c', (
-            "venv/bin/pip install Cython"
-        ), _env=copy.copy(base_env))
-
-        # Get environment variables for build (with CC/compiler set):
-        standard_recipe = CythonRecipe()
-        standard_recipe.ctx = ctx
-        # (note: following line enables explicit -lpython... linker options)
-        standard_recipe.call_hostpython_via_targetpython = False
-        recipe_env = standard_recipe.get_recipe_env(ctx.archs[0])
-        env = copy.copy(base_env)
-        env.update(recipe_env)
-
-        # Make sure our build package dir is available, and the virtualenv
-        # site packages come FIRST (so the proper pip version is used):
-        env["PYTHONPATH"] += ":" + ctx.get_site_packages_dir()
-        env["PYTHONPATH"] = os.path.abspath(join(
-            ctx.build_dir, "venv", "lib",
-            "python" + ctx.python_recipe.major_minor_version_string,
-            "site-packages")) + ":" + env["PYTHONPATH"]
-
-        # Install the manually specified requirements first:
-        if not modules:
-            info('There are no Python modules to install, skipping')
-        else:
-            info('Creating a requirements.txt file for the Python modules')
-            with open('requirements.txt', 'w') as fileh:
-                for module in modules:
-                    key = 'VERSION_' + module
-                    if key in environ:
-                        line = '{}=={}\n'.format(module, environ[key])
-                    else:
-                        line = '{}\n'.format(module)
-                    fileh.write(line)
+    for arch in ctx.archs:
```

Comment:
Minor: suggestion, this block below could also have its own function for semantic/readability and testing

Reply:
Addressed by moving the `for arch in ctx.archs` outside of the function